Snowflake.Data/Core/Session/CreateSessionTokens.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Collections.Generic;
+using Snowflake.Data.Core.Session;
+
+namespace Snowflake.Data.Core.Session
+{
+    internal class CreateSessionTokens: ICreateSessionTokens
+    {
+        internal long _timeout { get; set; }  = Timeout;
+        private const long Timeout = 30000; // 30 seconds as default
+        private readonly object _tokenLock = new object();
+        private readonly List<CreateSessionToken> _tokens = new List<CreateSessionToken>();
```

Comment:
In this code we obtain a lock twice in BeginCreate() and EndCreate() - it does not look too complicated to me. Actually having it as a list is simplification because we have such a nice method like RemoveAll() not available if it were ConcurrentBag for example.